runtime/src/accounts_db.rs

```diff
@@ -4230,6 +4228,19 @@ impl AccountsDb {
         result
     }
 
+    fn sort_storages(storages: &[SnapshotStorage]) -> Vec<&Arc<AccountStorageEntry>> {
+        let mut m1 = Measure::start("");
+        let mut result = storages
+            .iter()
+            .flatten()
+            .map(|item| (item.slot(), item))
```

Comment:
Can this be combined to [`.flat_map()`](https://doc.rust-lang.org/beta/std/iter/trait.Iterator.html#method.flat_map)?

Reply:
flat map says:
```
You can think of flat_map(f) as the semantic equivalent of mapping, and then flattening as in map(f).flatten().
```
In this case, I have flatten().map(). Maybe there is an iterator way to do this.

Reply:
I just put this in a pr to see if anything would break in the ci and to discuss with you. The current code doesn't take advantage of the fact that the storages would be sorted yet.

I think I may rework this more anyway. I need to guarantee later that if I chunk up an array of storages, that all storages for a slot are in a single chunk. This means there can never be 2 successive elements with the same slot.